### PR TITLE
Adding Docker context check for breeze

### PR DESCRIPTION
**Pull request by alexkruc**
Adding a docker context check for breeze to verify that we are running on the `default` context

Reference (From a Slack discussion): https://apache-airflow.slack.com/archives/CQ9QHSFQX/p1656525626619919 

image

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by alexkruc** [comment]
Do you think I should a message saying "Good docker context is used: {context}"?  
In the current implementation, in cases that the used context is the proper one, we don't print anything and just go forward with `breeze` :)

**Comment by potiuk** [comment]
Somehow the docker context check failed on ci :)

**Comment by alexkruc** [comment]
> Somehow the docker context check failed on ci :)

Yeah I see that :( 
I think that the subcommand `show` is not available there.
I'll try to find the context by another way :)
Is it ok to ping you for review when it will be ready?

**Comment by alexkruc** [comment]
@potiuk Tests are passing now :) 
Can you please review and merge if everything is ok? 🙏

File: dev/breeze/src/airflow_breeze/utils/docker_command_utils.py
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/utils/docker_command_utils.py]: I'd return into warning and immediately return from the function rather than print error. Generally as a rule "error" is reserved for some things that the user should fix to continue. 

We have this ADR written describing it: https://airflowsummit.org/sessions/2021/dynamic-security-roles-in-airflow-for-multi-tenancy/

In this case, if we cannot check docker, I think this might mean for example that for whatever reason docker "client' we have does not provide this information (but in this case I'd say things should work in general - so likely we can continue). 

The following warning is also a bit misleading - what does it mean "installed and running properly" ? The message should provide some helpful hints to the user - for example "properly (`docker info` should show Context: default")

File: dev/breeze/src/airflow_breeze/utils/docker_command_utils.py
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/utils/docker_command_utils.py:322]: as discused above - I think we should return here. Otherwise we are passing srdout of command that failed to parse the context - which might have strange results. What happen if stdout is None (I guess this wil happen if the process did not even manage to start) - but not sure about it. Regardless return here is better.

File: dev/breeze/src/airflow_breeze/utils/docker_command_utils.py
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/utils/docker_command_utils.py:331]: Nice.

File: dev/breeze/src/airflow_breeze/utils/docker_command_utils.py
Comment (potiuk) [inline dev/breeze/src/airflow_breeze/utils/docker_command_utils.py:335]: We should `sys.exit(1)` here. We know it makes no sense to continue. Save breeze the pain of trying fruitlessly to continue and kill it mercifully and immediately.

**Review by potiuk — CHANGES_REQUESTED** [review]
Some comments regarding communication and return/exit behaviour.

File: dev/breeze/src/airflow_breeze/utils/docker_command_utils.py
Comment (alexkruc) [reply dev/breeze/src/airflow_breeze/utils/docker_command_utils.py]: You're totally right :) 
We should return here because it's not making sense checking it further, and also I changed the message to be more actionable.

File: dev/breeze/src/airflow_breeze/utils/docker_command_utils.py
Comment (alexkruc) [reply dev/breeze/src/airflow_breeze/utils/docker_command_utils.py:322]: Added a return :) 
It's making a lot of sense.

File: dev/breeze/src/airflow_breeze/utils/docker_command_utils.py
Comment (alexkruc) [reply dev/breeze/src/airflow_breeze/utils/docker_command_utils.py:335]: I was torn a bit about this, because it depends on the flow of `breeze`, for example, when you run `breeze build-image`, continuing past this is not making sense. But, when running just `breeze` to connect to a container, it works even with the wrong context, it's just using an old image instead of an updated one.
BUT
Running with an outdated image can lead to strange behaviors and errors that will just fail the CI, so I've added the `sys.exit(1)`.

Thanks for this comment :)

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Nice one !

**Comment by potiuk** [comment]
:D

image

**Comment by alexkruc** [comment]
> Nice one !

Thank you :)
I see that you approved it, can we also merge this change? Or are you waiting for another reviewer or something else? I don't have write access to the repo as I'm not a committer (hopefully one day :) )

**Comment by potiuk** [comment]
Yep. I was waiting for CI to complete the checks :).

**Pull request merged**
